### PR TITLE
Foundations/CSS Foundations: format headings and structure according to style guide

### LINKED ISSUE #24339: Style Guide - Foundations/CSS Foundations: Headings and Section Placement
# Foundations/CSS Foundations

This issue is part of the Style Guide Community Project, a way of offering members of The Odin Project community a chance to contribute to our curriculum. Users who wish to take part and contribute to any Style Guide issues should have thoroughly read the [layout style guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md) and [The Odin Project contributing guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md).
  
## Description of Issue

This issue ensures that headings are properly formatted and that all sections are placed in the correct order.

Each applicable lesson for this issue should meet the following bulleted requirements:

- The main headings use the Markdown `###` syntax.
- Sub-headings that are not inline use the Markdown `####` syntax.
- Sub-headings that are inline use the Markdown `**sub-heading**` (i.e. bold) syntax.
- Both headings and sub-headings use Wikipedia style Title Case.
- There is a newline before and after headings and non-inline sub-headings.
- The order of existing sections follows the below format:
	- Introduction
	- Lesson Overview
	- Custom Sections (lessons) / Pre-Assignment Sections (projects)
	- Assignment
	- Practice (lessons) / Post-Assignment Sections (projects)
	- Knowledge Check
	- Additional Resources
- Additionally, there should be an empty newline at the end of the file.
 
## Acceptance Criteria

- [ ] The following lessons are updated according to the issue description above:
	- [ ] [CSS Foundations](https://www.theodinproject.com/paths/foundations/courses/foundations/lessons/css-foundations)
            - Remove the code backticks from headings
	- [ ] [Inspecting HTML and CSS](https://www.theodinproject.com/paths/foundations/courses/foundations/lessons/inspecting-html-and-css)
	- [ ] [The Box Model](https://www.theodinproject.com/paths/foundations/courses/foundations/lessons/the-box-model)
	- [ ] [Block and Inline](https://www.theodinproject.com/paths/foundations/courses/foundations/lessons/block-and-inline)

### PR DESCRIPTION
**Pull request by chri-ss**
Complete the following REQUIRED checkboxes:

-   [ x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [ x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`

Complete the following checkboxes ONLY IF they are applicable to your PR. You can complete them later if they are not currently applicable:
-   [x ] I have previewed all lesson files included in this PR with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure the Markdown content is formatted correctly
-   [ x] I have ensured all lesson files included in this PR follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

<hr>

**1. Because:**

Closes #24339

**2. This PR:**

- Formatted headings and sub-headings in the required files according to style guidelines. 
- Re-ordered sections to according to style guide

**3. Additional Information:**

### CODE REVIEW

File: foundations/html_css/css-foundations.md [hunk 1]
```diff
@@ -216,7 +217,7 @@ There's really no limit to how many combinators you can add to a rule, so `.one
 
 There are some CSS properties that you're going to be using all the time, or at the very least more often than not. We're going to introduce you to several of these properties, though this is by no means a complete list. Learning the following properties will simply be enough to help get you started.
 
-#### `color` and `background-color`
+#### color and background-color
```
Comment (thatblindgeye) [inline foundations/html_css/css-foundations.md:220]:
```suggestion
#### Color and Background-Color
```

File: foundations/html_css/css-foundations.md [hunk 2]
```diff
@@ -235,7 +236,7 @@ p {
 
 Take a quick look at [CSS Legal Color Values](https://www.w3schools.com/cssref/css_colors_legal.asp) to see how you can adjust the opacity of these colors by adding an alpha value.
 
-#### Typography Basics and `text-align`
+#### Typography Basics and text-align
```
Comment (thatblindgeye) [inline foundations/html_css/css-foundations.md:239]:
```suggestion
#### Typography Basics and Text-Align
```

**Review by thatblindgeye — CHANGES_REQUESTED** [review]
Thanks for making these updates! Had a few change requests below. Additionally:

- We don't need a newline at the start of the files, so we can remove that so the first line is `### Introduction` in these files
- Can you also ensure the headings in the Inspecting HTML and CSS lesson follow the Wikipedia style title case?

File: foundations/html_css/css-foundations.md (hunk 2 above)
Comment (chri-ss) [reply foundations/html_css/css-foundations.md]: Hey, I believe I fixed these issues. May I resolve the conversation?

**Review by zachmmeyer — APPROVED** [review]
Thank you!

**Pull request merged**